pyro/distributions/zero_inflated.py

```diff
@@ -41,9 +51,17 @@ def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
 
-        gate, value = broadcast_all(self.gate, value)
-        log_prob = (-gate).log1p() + self.base_dist.log_prob(value)
-        log_prob = torch.where(value == 0, (gate + log_prob.exp()).log(), log_prob)
+        if 'gate' in self.__dict__:
+            gate, value = broadcast_all(self.gate, value)
+            log_prob = (-gate).log1p() + self.base_dist.log_prob(value)
+            log_prob = torch.where(value == 0, (gate + log_prob.exp()).log(), log_prob)
+        else:
+            gate_logits, value = broadcast_all(self.gate_logits, value)
+            log_gate = -softplus(-gate_logits)
+            log_one_minus_gate = log_gate - gate_logits
+            log_prob = log_one_minus_gate + self.base_dist.log_prob(value)
+            zero_log_prob = softplus(log_prob - log_gate) + log_gate
```

Comment:
We can save one operator here if we compute `log_prob_minus_log_gate` first then plus with loggate to compute logprob.

Reply:
great, thanks! like this right?

Reply:
yup